src/index.js

```diff
@@ -194,8 +195,61 @@ const HMMediaWeightSidebar = () => {
 		};
 	} );
 
+	const insertIframe = ( url, width = '1600px', height = '800px' ) => {
+		const iframeExists = document.getElementById( 'post-preview-iframe' );
+
+		if ( ! iframeExists ) {
+			const iframe = document.createElement( 'iframe' );
+			iframe.src = url;
+			iframe.width = width;
+			iframe.height = height;
+			iframe.id = 'post-preview-iframe';
+			iframe.style.display = 'none';
+			document.body.appendChild( iframe );
+		}
+	}
+
+	const addQueryString = ( url, params ) => {
+		const urlObject = new URL( url );
+		const searchParams = new URLSearchParams( urlObject.search );
+
+		for ( const key in params ) {
+			if ( params.hasOwnProperty( key ) ) {
+				searchParams.set( key, params[key] );
+			}
+		}
+
+		urlObject.search = searchParams.toString();
+
+		return urlObject.toString();
+	}
+
+	const newParams = {
+		mediaWeight: '1'
+	};
+	const iframeURL = addQueryString( previewPostLink, newParams );
+
+	useEffect( () => {
+		window.addEventListener( 'message', ( event ) => {
+			const receivedEntries = event.data;
+			// eslint-disable-next-line no-console
+			console.log( receivedEntries );
+		} );
+	}, [ iframeURL ] );
```

Comment:
When you bind events like this in a `useEffect()` you should also handle unbinding them in the `useEffect()` return statement to avoid binding multiple times:

```suggestion
	useEffect( () => {
		const listener = window.addEventListener( 'message', ( event ) => {
			const receivedEntries = event.data;
			// eslint-disable-next-line no-console
			console.log( receivedEntries );
			// You can call a useState callback here to get the data into the component
		} );
		return () => {
			window.removeEventListener( listener );
		};
	}, [] );
```

Reply:
I implemented your suggestion, thank you.